src/query/api/v1/handler/prometheus/common.go

```diff
@@ -304,6 +304,35 @@ func renderDefaultTagCompletionResultsJSON(
 	return jw.Close()
 }
 
+// RenderListTagResultsJSON renders list tag results to json format.
+func RenderListTagResultsJSON(
```

Comment:
can we rely on `json.Marshal` with the appropriate struct instead of defining our own?

Reply:
Yeah, it's an issue with a bunch of our stuff; called out removing json.Marshal in #1590

Reply:
Added a response on that issue, for any endpoints where we return a lot of data we should continue using our JSON writers:
> One thing interesting about the JSON writers, the major reason in the past we did that is because the default JSON marshaller allocates a lot, however if we have JSON write methods for bytes/strings/whatever other concrete types we can avoid allocations when returning responses (which is what we have today with the current common library streamjson writer).

Reply:
Yeah was going to write some quick benchmarks when it came time to consider swapping it out; I think our initial thinking here was if we're using this writer internally there's a good reason